### PR TITLE
DataGrid: update the 'scrolling.mode' description

### PR DESCRIPTION
**Pull request by vladaskorohodova**

### CODE REVIEW

File: api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md [hunk 1]
```diff
@@ -3,29 +3,38 @@ id: dxDataGrid.Options.scrolling.mode
...
+A user scrolls a single page only. Scrolling is available only if [all the page's rows](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#pageSize') do not fit into the UI component's height. In this mode, the [pager](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/pager '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/pager/') performs the main navigation and scrolling is auxiliary. If you disable the [paging](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/paging '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/'), the UI component loads all rows simultaneously. This action can reduce the UI component's performance. In this case, use another scrolling mode.
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md]:
```suggestion
A user scrolls a single page only. Scrolling is available only if [all rows of a page](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#pageSize') do not fit within the UI component's height. In this mode, the [pager](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/pager '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/pager/') performs the main navigation and scrolling is auxiliary. If you disable the [paging](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/paging '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/') feature, the UI component loads all rows simultaneously. This action can reduce the UI component's performance. In this case, use another scrolling mode.
```

File: api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md [hunk 2]
```diff
@@ -3,29 +3,38 @@ id: dxDataGrid.Options.scrolling.mode
...
+The component loads pages when they get into the viewport and removes them once they leave it. In this mode, users can move to any page instantly. If the component takes time to load and render rows, they display gray boxes. [Rendering optimization](/concepts/05%20UI%20Components/DataGrid/10%20Enhance%20Performance%20on%20Large%20Datasets/030%20Rendering%20Optimization.md '/Documentation/Guide/UI_Components/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization') can reduce rendering time and remove the gray boxes. Alternatively, you can disable the **scrolling**.[renderAsync](/api-reference/10%20UI%20Components/GridBase/1%20Configuration/scrolling/renderAsync.md '{basewidgetpath}/Configuration/scrolling/#renderAsync') property, but this can impact performance.
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md]: 1. `The component loads pages when they get into the viewport and removes them once they leave it.`

a) Are you sure "pages" is correct? Or do you mean "rows" (or "records"?)? Just checking to make sure.
(My understanding is that virtual scrolling allows you to scroll through thousands of records on a single page. Is my understanding incorrect?)

b) Can we change "when they get into the viewport" to "while they are visible within the viewport"?

2. `[Rendering optimization](/concepts/05%20UI%20Components/DataGrid/10%20Enhance%20Performance%20on%20Large%20Datasets/030%20Rendering%20Optimization.md '/Documentation/Guide/UI_Components/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization') can reduce rendering time and remove the gray boxes.`

Can we change "rendering time" to "render time"?

File: api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md [hunk 3]
```diff
@@ -3,29 +3,38 @@ id: dxDataGrid.Options.scrolling.mode
...
 
 - **Infinite**      
-Each next page is loaded once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually from the first to the last page. Row selection in **infinite** and **virtual** scroll mode with Shift+Click does not work in the following cases: 
+The component loads each next once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually from the first to the last page. Row selection in **infinite** and **virtual** scroll mode with Shift+Click does not work in the following cases:
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md]: `The component loads each next once the scrollbar reaches the end of its scale.`

What do you mean by "The component loads each next"? What does it load next?

File: api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md [hunk 4]
```diff
@@ -3,29 +3,38 @@ id: dxDataGrid.Options.scrolling.mode
...
 
 - **Infinite**      
-Each next page is loaded once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually from the first to the last page. Row selection in **infinite** and **virtual** scroll mode with Shift+Click does not work in the following cases: 
+The component loads each next once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually from the first to the last page. Row selection in **infinite** and **virtual** scroll mode with Shift+Click does not work in the following cases: 
     - Deferred selection is enabled.
     - Rows grouping is enabled.
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md]:
```suggestion
    - Row grouping is enabled.
```

File: api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md (hunk 2 above)
Comment (vladaskorohodova) [reply api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md]: 1. a) Customers often think that virtual scrolling only affects how rows are rendered. However, it is very important that virtual/infinite scrolling always loads data by pages. Even if paging.enabled is set to false, this setting will be ignored.

Should I write about it? I just thought that mentioning 'pages' and not 'rows' would be enough.

File: api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md (hunk 3 above)
Comment (vladaskorohodova) [reply api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md]: I simply forgot the word "page", thanks for pointing that out.

File: api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md (hunk 2 above)
Comment (albertov05) [reply api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md]: I think the way you wrote it ("The component loads pages...") is sufficient to emphasize that virtual scrolling loads data by page. Thanks for the clarification.

File: api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md [hunk 5]
```diff
@@ -3,29 +3,38 @@ id: dxDataGrid.Options.scrolling.mode
...
 
 - **Infinite**      
-Each next page is loaded once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually from the first to the last page. Row selection in **infinite** and **virtual** scroll mode with Shift+Click does not work in the following cases: 
+The component loads each next page once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually from the first to the last page. Row selection in **infinite** and **virtual** scroll mode with Shift+Click does not work in the following cases:
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md]:
```suggestion
The component loads the next page once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually from the first to the last page. Row selection in **infinite** and **virtual** scroll mode with Shift+Click does not work in the following cases:
```

File: api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md (hunk 3 above)
Comment (albertov05) [reply api-reference/10 UI Components/dxDataGrid/1 Configuration/scrolling/mode.md]: Thanks, Vlada. I added a suggestion to update your new version (change "each next page" to "the next page").

**Pull request merged**